gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/EhrExtractMessageHandler.java

```diff
@@ -242,6 +244,29 @@ private PatientAttachmentLog buildPatientAttachmentLogFromAttachment(
         PatientMigrationRequest migrationRequest,
         InboundMessage.Attachment attachment) throws ParseException {
 
+        String description = attachment.getDescription();
```

Comment:
It's a shame we have this behaviour duplicated across both this class and `InlineAttachment`.

Reply:
I'm a little surprised that we don't have any unit tests which needed changing for `EhrExtractMessageHandler` class.

Reply:
After doing some digging, I suspect that moving all the logic into `InlineAttachment` feels like a sensible route forward for reducing the duplication.

Reply:
I've had another look at this, and I'm not sure how we could refactor it into a nice way. I suspect the reason for this is I don't fully understand the requriements.  @benhession might be able to help 😢

Reply:
I've left some comments below that relate to this, we could use the `XMLParseUtilService` to reduce the regex duplication. The attachment log is keeping track of the attachments that are expected. The `InlineAttachment` class is used to represent an attachment or part of an attachment that has been received.